pkg/apis/certmanager/v1/types_certificate.go

```diff
@@ -437,3 +442,15 @@ const (
 	// It will be removed by the 'issuing' controller upon completing issuance.
 	CertificateConditionIssuing CertificateConditionType = "Issuing"
 )
+
+// CertificateSecretTemplate defines the default labels and annotations
+// to be copied to the Kubernetes Secret resource named in `CertificateSpec.secretName`.
+type CertificateSecretTemplate struct {
+	// Annotations is a key value map to be copied to the target Kubernetes Secret.
+	// +optional
+	Annotations map[string]string `json:"annotations,omitempty"`
+
+	// Labels is a key value map to be copied to the target Kubernetes Secret.
+	// +optional
+	Labels map[string]string `json:"labels,omitempty"`
```

Comment:
@wallrj 

I couldn't find a simple way to use CRD validations that satisfies the same rules we have in `ValidateAnnotations()` (from `k8s.io/apimachinery/pkg/api/validation`) and `ValidateLabels()`(from `k8s.io/apimachinery/pkg/apis/meta/v1/validation`), so I ended up using these functions.

Could you please review if this is, kind of, what is expected?

Reply:
These functions seem to do the right thing, but I think the validation should happen in the validating webhook, as Richard says, so we fail faster. The `Certificate` fields are validated [here](https://github.com/jetstack/cert-manager/blob/master/pkg/internal/apis/certmanager/validation/certificate.go)

Reply:
thanks @irbekrm I will update the PR soon

Reply:
Thank you @jonathansp ! We also had a brief chat about https://github.com/jetstack/cert-manager/pull/3828#issuecomment-887298287 at a team standup earlier (we have a code freeze coming up and this looks like a much wanted feature) and thought that perhaps we could use a separate annotation on the secret to store the information about which labels/annotations have been added from the secrets template specifically so that they can be removed/deleted as well as added. Is that something you'd be interested to work on? I think this PR in any case already does a good job!

Reply:
> These functions seem to do the right thing, but I think the validation should happen in the validating webhook, as Richard says, so we fail faster. The `Certificate` fields are validated [here](https://github.com/jetstack/cert-manager/blob/master/pkg/internal/apis/certmanager/validation/certificate.go)

Hi @irbekrm I have moved these validations to the validation package. Much better now 😄

Reply:
> Thank you @jonathansp ! We also had a brief chat about [#3828 (comment)](https://github.com/jetstack/cert-manager/pull/3828#issuecomment-887298287) at a team standup earlier (we have a code freeze coming up and this looks like a much wanted feature) and thought that perhaps we could use a separate annotation on the secret to store the information about which labels/annotations have been added from the secrets template specifically so that they can be removed/deleted as well as added. Is that something you'd be interested to work on? I think this PR in any case already does a good job!

I'd be happy to work on it. Should it be part of this PR or better to have a new one? Do you guys have an idea of the name/content? something like a JSON string?

Reply:
Hi @irbekrm I've created https://github.com/jetstack/cert-manager/issues/4292 to keep this tracked as @wallrj suggested.